cmd/xl-v1-metadata.go

```diff
@@ -17,20 +17,61 @@
 package cmd
 
 import (
+	"bytes"
+	"encoding/hex"
 	"encoding/json"
 	"errors"
 	"path"
-	"runtime"
 	"sort"
 	"sync"
 	"time"
+
+	"github.com/aead/poly1305"
+	"github.com/minio/minio/pkg/bitrot"
+	sha256 "github.com/minio/sha256-simd"
+	"golang.org/x/crypto/blake2b"
 )
 
 const (
 	// Erasure related constants.
 	erasureAlgorithmKlauspost = "klauspost/reedsolomon/vandermonde"
 )
 
+// DefaultBitrotAlgorithm is the default algorithm used for bitrot protection
+const DefaultBitrotAlgorithm = bitrot.Poly1305
```

Comment:
The reference implementation will **not** be fast for ARM since the go compiler has some way to go for ARM.
And especially also because the SHA256 uses the crypto instructions to speed up hashing speed to BLAKE2 like performance.
We need to get some benchmark numbers in order to make a decision